### ISSUE TITLE
Pacman sometimes escapes death from the Ghost

### ISSUE THREAD

**Issue by shivangg** [pesto-students/batch5-pacman#85]
![pacmanDoesntDie](https://user-images.githubusercontent.com/10586637/54963298-45697400-4f8e-11e9-802e-0e495f3e8fec.gif)

As seen in GIF, the ghosts were not able to kill the pacman.

**Comment by shivangg** [comment]
After fiddling with this issue for sometime, the reason i found for this behaviour is that **when the distance between approaching pacman and ghost is even, they will go pass each other.**

Eg: Take this 1D example

Frame 1: `G00P`
Frame 2: `0GP0`
Frame 3: `0PG0`
Frame 4: `P00G`

As it can be seen, under no case, `P` and `G` are at the same cell. P gets killed only when distance in between is odd.

To solve this issue, put the `dieIfOnGhost` in the intermediate step. After moving the ghosts, when moving the pacman, first check for ghost using the `dieIfOnGhost`. This way even distance can be handled.

Note: This might encourage one to remove the previous `dieIfOnGhost` that gets called after the movement of both the entitites. Removing that will lead to failing deaths in case of odd distance.

TLDR: 
1. Use `dieIfOnGhost` in `movePacman()` to handle even distance.
2. use `dieIfOnGhost` after `moveGhosts()` + `movePacman()` to handle odd distance.

![image](https://user-images.githubusercontent.com/10586637/54965144-3afea880-4f95-11e9-84da-c5013b8f0f7a.png)
